packages/javascript-sdk/src/fr-webauthn/index.ts

```diff
@@ -273,19 +290,33 @@ abstract class FRWebAuthn {
     try {
       const clientDataJSON = arrayBufferToString(credential.response.clientDataJSON);
       const assertionResponse = credential.response as AuthenticatorAssertionResponse;
-      const authenticatorData = new Int8Array(assertionResponse.authenticatorData).toString();
+      const authenticatorData = new Int8Array(
+        assertionResponse.authenticatorData,
+      ).toString() as AttestationType;
       const signature = new Int8Array(assertionResponse.signature).toString();
 
       // Current native typing for PublicKeyCredential does not include `userHandle`
       // eslint-disable-next-line
       // @ts-ignore
       const userHandle = arrayBufferToString(credential.response.userHandle);
 
-      let stringOutput = `${clientDataJSON}::${authenticatorData}::${signature}::${credential.id}`;
+      let stringOutput =
+        `${clientDataJSON}::${authenticatorData}::${signature}::${credential.id}` as DeviceName<
+          string,
+          AttestationType,
+          PublicKeyCredential
+        >;
       // Check if Username is stored on device
       if (userHandle) {
         stringOutput = `${stringOutput}::${userHandle}`;
+        return stringOutput as DeviceName<
+          string,
+          AttestationType,
+          PublicKeyCredential,
+          typeof userHandle
```

Comment:
I suppose this will always be string, but if `userHandle` is ever a more strict type then we could leverage that here.

Reply:
Right but one could pass “username” and in typescript 5 we could leverage a const generic which would make the type “username” and not string so it could be used more strictly